packages/libs/eda/src/lib/map/analysis/mapTypes/shared.tsx

```diff
@@ -547,17 +549,36 @@ export const GLOBAL_VIEWPORT = {
  */
 
 const noDataPatterns = [
-  /did not contain any data/, // comes from map-markers endpoint
   /Could not generate continuous variable metadata/, // comes from continuous-variable-metadata endpoint
 ];
 
 export function isNoDataError(error: unknown) {
+  if (error instanceof Error && error.name === 'NoDataError') return true;
   return noDataPatterns.some((pattern) => String(error).match(pattern));
 }
 
-export const noDataErrorMessage = (
+const noDataErrorMessage = (
```

Comment:
Could possibly rename this `noDataLegendMessage`? And move it down next to the function that uses it.